packages/quantic/force-app/main/default/lwc/quanticResultChildren/__tests__/quanticResultChildren.test.js

```diff
@@ -0,0 +1,260 @@
+  toggleButton: '.folded-result__toggle-button lightning-button',
+  placeholder: 'c-quantic-placeholder',
+  noMoreChildrenMessage: '.result-children__no-more-children-message',
+};
+
+const exampleFoldedResultOne = {
+  result: {
+    uniqueId: '123',
+    title: 'example result one',
+  },
+  children: [],
+};
+const exampleFoldedResultTwo = {
+  result: {
+    uniqueId: '456',
+    title: 'example result two',
+  },
+  children: [],
+};
+const exampleEngineId = 'example engine id';
+const exampleTemplateId = 'example template id';
+const foldedCollection = {
+  result: {
+    uniqueId: '789',
+    title: 'example parent result',
+  },
+  children: [exampleFoldedResultOne],
+  isLoadingMoreResults: false,
+  moreResultsAvailable: true,
+};
+const foldedCollectionWithAllResultsLoaded = {
+  ...foldedCollection,
+  children: [exampleFoldedResultOne, exampleFoldedResultTwo],
+  isLoadingMoreResults: false,
+  moreResultsAvailable: false,
+};
+
+const defaultOptions = {
+  engineId: exampleEngineId,
+  templateId: exampleTemplateId,
+  foldedResultListController: mockedFoldedResultListController,
+  foldedCollection: foldedCollection,
+};
+
+function createTestComponent(options = defaultOptions) {
+  const element = createElement('c-quantic-result-children', {
+    is: QuanticResultChildren,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+function cleanup() {
+  jest.clearAllMocks();
+  // The jsdom instance is shared across test cases in a single file so reset the DOM
+  while (document.body.firstChild) {
+    document.body.removeChild(document.body.firstChild);
+  }
+}
+
+function expectProperChildResultsDisplay(element, collection) {
+  const childResults = element.shadowRoot.querySelectorAll(
+    selectors.quanticResult
+  );
+  expect(childResults.length).toBe(collection.children.length);
+  collection.children.forEach((foldedResult, index) => {
+    expect(childResults[index].engineId).toBe(exampleEngineId);
+    expect(childResults[index].templateId).toBe(exampleTemplateId);
+    expect(childResults[index].result).toEqual(foldedResult.result);
+    expect(childResults[index].foldedCollection).toEqual(foldedResult);
+  });
+}
+
+async function clickFoldedResultToggleButton(element, label) {
+  const foldedResultToggleButton = element.shadowRoot.querySelector(
+    selectors.toggleButton
+  );
+  expect(foldedResultToggleButton).not.toBeNull();
+  expect(foldedResultToggleButton.label).toBe(label);
+  foldedResultToggleButton.click();
+  await flushPromises();
+}
+
+describe('c-quantic-result-children', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('when there are no child results available in the collection', () => {
```

Comment:
selecting the right template is ran at the level of the Quantic Result component so we don't have such a test for the QuanticResultChilldren component..
 we can add unit tests to the Quantic Result to cover this but for now this will be covered by the Cypress E2E.